pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -492,35 +503,38 @@ func (c *Controller) syncCheckStates(ctx context.Context, wl *kueue.Workload, ch
 
 			prFailed := apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Failed)
 			prProvisioned := apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Provisioned)
-			log.V(3).Info("Synchronizing admission check state based on provisioning request", "wl", klog.KObj(wl), "check", check, "prName", pr.Name, "failed", prFailed, "accepted", prProvisioned)
+			prAccepted := apimeta.IsStatusConditionTrue(pr.Status.Conditions, autoscaling.Accepted)
+			log.V(3).Info("Synchronizing admission check state based on provisioning request", "wl", klog.KObj(wl),
+				"check", check,
+				"prName", pr.Name,
+				"failed", prFailed,
+				"provisioned", prProvisioned,
+				"accepted", prAccepted)
 
 			switch {
 			case prFailed:
 				if checkState.State != kueue.CheckStateRejected {
 					if attempt := getAttempt(ctx, pr, wl.Name, check); attempt <= MaxRetries {
 						// it is going to be retried
 						message := fmt.Sprintf("Retrying after failure: %s", apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message)
-						updated = updated || checkState.State != kueue.CheckStatePending || checkState.Message != message
-						checkState.State = kueue.CheckStatePending
-						checkState.Message = message
+						updated = updated || updateCheckState(&checkState, kueue.CheckStatePending) || updateCheckMessage(&checkState, message)
 					} else {
 						updated = true
 						checkState.State = kueue.CheckStateRejected
 						checkState.Message = apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Failed).Message
 					}
 				}
 			case prProvisioned:
-				if checkState.State != kueue.CheckStateReady {
+				if updateCheckState(&checkState, kueue.CheckStateReady) {
 					updated = true
-					checkState.State = kueue.CheckStateReady
 					// add the pod podSetUpdates
 					checkState.PodSetUpdates = podSetUpdates(wl, pr)
 				}
 			default:
-				if checkState.State != kueue.CheckStatePending {
-					updated = true
-					checkState.State = kueue.CheckStatePending
-				}
+				updated = updated || updateCheckState(&checkState, kueue.CheckStatePending)
+			}
+			if prAccepted && !prFailed {
+				updated = updated || updateCheckMessage(&checkState, apimeta.FindStatusCondition(pr.Status.Conditions, autoscaling.Provisioned).Message)
```

Comment:
Can this be a part of the switch statement for consistency? Could you add a comment explaining this case in fact covers `Provisioned=false` condition?